sstables/mx/bsearch_clustered_cursor.hh

```diff
@@ -526,6 +526,22 @@ public:
         });
     }
 
+    future<position_range> full_clustering_range() override {
+        if (_blocks_count == 0) {
+            co_return position_range::all_clustered_rows();
+        }
+
+        auto first_block = co_await _promoted_index.get_block(0, _trace_state);
+        auto last_block = co_await _promoted_index.get_block(_blocks_count - 1, _trace_state);
```

Comment:
These two could be done in parallel.  I think we have enough concurrency when opening sstables, so probably not needed.

Reply:
correct. I usually avoid parallelism on higher layer (if not strictly needed), as lower layers are doing lots of parallelism already using write behind, read ahead, etc. Parallelism on higher layers makes more sense if they're serving different things. Otherwise, better to serialize.